src/transformers/trainer.py

```diff
@@ -287,7 +287,7 @@ def get_optimizers(
         If you want to use something else, you can pass a tuple in the Trainer's init,
         or override this method in a subclass.
         """
-        if self.optimizers is not None:
+        if self.optimizers is not Noneself.eval:
```

Comment:
Be careful here!

Reply:
must have been a middle click by mistake… Shows how tests are important.
I'll probably add some loggers tests in the future in a separate PR.